### ISSUE TITLE
s3prl upstream has no model attribute

### ISSUE THREAD

**Issue by jwrh** [espnet/espnet#5002]
**Describe the bug**
There seems to be a bug in the implementation of `s3prl.py` (`espnet/espnet2/asr/frontend/s3prl.py`). The implementation starting from line 54 to line 61 would never get executed because `upstream`, which is a instance of `S3PRLUpstream`, does not have a `model` attribute. Instead, it does have an attribute called `upstream`, which has a `model` attribute. Therefore, currently if we are training asr with a `Wav2Vec2Model` or `HubertModel`, the `if` statement on line 54 to always evaluate to False and we will not be able to set `upstream.model.encoder.layerdrop = 0.0`, causing interference in model performance.

Thus I believe this to be a bug, and could be fixed by replacing `upstream.model.__class__.__name__ ` to `upstream.upstream.model.__class__.__name__ `.

**Basic environments:**
- OS information: Linux 5.4.0-137-generic #154-Ubuntu SMP Thu Jan 5 17:03:22 UTC 2023 x86_64
- python version: `3.9.16 (main, Mar  1 2023, 18:22:10)  [GCC 11.2.0]`
- espnet version: `espnet 202301`
- pytorch version: `pytorch 1.13.1`
- Git hash: `3ad994f931219649e0113c5926926c339d777118`
  - Commit date: `Sat Mar 4 23:47:37 2023 -0500`
 

**Environments from `torch.utils.collect_env`:**
e.g.,
```
Collecting environment information...
PyTorch version: 1.4.0
Is debug build: No
CUDA used to build PyTorch: 10.0

OS: CentOS Linux release 7.5.1804 (Core)
GCC version: (GCC) 7.2.0
CMake version: version 2.8.12.2

Python version: 3.7
Is CUDA available: Yes
CUDA runtime version: 10.0.130
GPU models and configuration:
GPU 0: TITAN RTX
GPU 1: TITAN RTX
GPU 2: TITAN RTX
GPU 3: TITAN RTX

Nvidia driver version: 440.33.01
cuDNN version: Could not collect

Versions of relevant libraries:
[pip3] numpy==1.18.5
[pip3] pytorch-ranger==0.1.1
[pip3] pytorch-wpe==0.0.0
[pip3] torch==1.4.0
[pip3] torch-complex==0.1.1
[pip3] torch-optimizer==0.0.1a14
[pip3] torchaudio==0.4.0
[pip3] warprnnt-pytorch==0.1
[conda] blas                      1.0                         mkl
[conda] mkl                       2020.1                      217
[conda] mkl-service               2.3.0            py37he904b0f_0
[conda] mkl_fft                   1.1.0            py37h23d657b_0
[conda] mkl_random                1.1.1            py37h0573a6f_0
[conda] pytorch                   1.4.0           py3.7_cuda10.0.130_cudnn7.6.3_0    pytorch
[conda] pytorch-ranger            0.1.1                    pypi_0    pypi
[conda] pytorch-wpe               0.0.0                    pypi_0    pypi
[conda] torch-complex             0.1.1                    pypi_0    pypi
[conda] torch-optimizer           0.0.1a14                 pypi_0    pypi
[conda] torchaudio                0.4.0                    pypi_0    pypi
[conda] warprnnt-pytorch          0.1                      pypi_0    pypi
```
You can obtain them by the following command

```
cd <espnet-root>/tools
. ./activate_python.sh
python3 -m torch.utils.collect_env
```

**Task information:**
 - Task: ASR
 - Recipe: Mini_an4
 - ESPnet2

**To Reproduce**
1. move to a recipe directory, e.g., `cd egs2/mini_an4/asr1`
2. execute `run.sh` with specific arguments, e.g., `run.sh --stage 11 --stop_stage 11` with a model specification that includes wav2vec2 model

**Error logs**

```
# python3 -m espnet2.bin.asr_train --use_preprocessor true --bpemodel data/en_token_list/bpe_unigram30/bpe.model --token_type bpe --token_list data/en_token_list/bpe_unigram30/tokens.txt --non_linguistic_symbols none --cleaner none --g2p none --valid_data_path_and_name_and_type dump/raw/train_dev/wav.scp,speech,sound --valid_shape_file exp/asr_stats_raw_en_bpe30/valid/speech_shape --resume true --ignore_init_mismatch false --fold_length 80000 --output_dir exp/asr_test3_raw_en_bpe30 --config conf/test3.yaml --frontend_conf fs=16k --normalize=global_mvn --normalize_conf stats_file=exp/asr_stats_raw_en_bpe30/train/feats_stats.npz --train_data_path_and_name_and_type dump/raw/train_nodev/wav.scp,speech,sound --train_shape_file exp/asr_stats_raw_en_bpe30/train/speech_shape --fold_length 150 --train_data_path_and_name_and_type dump/raw/train_nodev/text,text,text --train_shape_file exp/asr_stats_raw_en_bpe30/train/text_shape.bpe --valid_data_path_and_name_and_type dump/raw/train_dev/text,text,text --valid_shape_file exp/asr_stats_raw_en_bpe30/valid/text_shape.bpe --ngpu 1 --multiprocessing_distributed True 
# Started at Mon Mar 13 11:57:00 EDT 2023
#
/project/ocean/junweih/sandbox/espnet/tools/anaconda/envs/espnet/lib/python3.9/site-packages/torch/cuda/__init__.py:497: UserWarning: Can't initialize NVML
  warnings.warn("Can't initialize NVML")
/project/ocean/junweih/sandbox/espnet/tools/anaconda/envs/espnet/bin/python3 /project/ocean/junweih/sandbox/espnet/espnet2/bin/asr_train.py --use_preprocessor true --bpemodel data/en_token_list/bpe_unigram30/bpe.model --token_type bpe --token_list data/en_token_list/bpe_unigram30/tokens.txt --non_linguistic_symbols none --cleaner none --g2p none --valid_data_path_and_name_and_type dump/raw/train_dev/wav.scp,speech,sound --valid_shape_file exp/asr_stats_raw_en_bpe30/valid/speech_shape --resume true --ignore_init_mismatch false --fold_length 80000 --output_dir exp/asr_test3_raw_en_bpe30 --config conf/test3.yaml --frontend_conf fs=16k --normalize=global_mvn --normalize_conf stats_file=exp/asr_stats_raw_en_bpe30/train/feats_stats.npz --train_data_path_and_name_and_type dump/raw/train_nodev/wav.scp,speech,sound --train_shape_file exp/asr_stats_raw_en_bpe30/train/speech_shape --fold_length 150 --train_data_path_and_name_and_type dump/raw/train_nodev/text,text,text --train_shape_file exp/asr_stats_raw_en_bpe30/train/text_shape.bpe --valid_data_path_and_name_and_type dump/raw/train_dev/text,text,text --valid_shape_file exp/asr_stats_raw_en_bpe30/valid/text_shape.bpe --ngpu 1 --multiprocessing_distributed True
[islpc50] 2023-03-13 11:57:08,372 (asr:489) INFO: Vocabulary size: 30
[islpc50] 2023-03-13 11:57:09,198 (download:170) INFO: Requesting URL: https://huggingface.co/s3prl/converted_ckpts/resolve/main/wav2vec_small.pt
[islpc50] 2023-03-13 11:57:09,211 (download:181) INFO: Using URL's local file: hub/aa064e275fe0123a0e1b515f2341bbe4408368510d91d0a6816f2822a6e5acdd.wav2vec_small.pt
Traceback (most recent call last):
  File "/project/ocean/junweih/sandbox/espnet/tools/anaconda/envs/espnet/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/project/ocean/junweih/sandbox/espnet/tools/anaconda/envs/espnet/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/project/ocean/junweih/sandbox/espnet/espnet2/bin/asr_train.py", line 23, in <module>
    main()
  File "/project/ocean/junweih/sandbox/espnet/espnet2/bin/asr_train.py", line 19, in main
    ASRTask.main(cmd=cmd)
  File "/project/ocean/junweih/sandbox/espnet/espnet2/tasks/abs_task.py", line 1054, in main
    cls.main_worker(args)
  File "/project/ocean/junweih/sandbox/espnet/espnet2/tasks/abs_task.py", line 1164, in main_worker
    model = cls.build_model(args=args)
  File "/project/ocean/junweih/sandbox/espnet/espnet2/tasks/asr.py", line 495, in build_model
    frontend = frontend_class(**args.frontend_conf)
  File "/project/ocean/junweih/sandbox/espnet/espnet2/asr/frontend/s3prl.py", line 61, in __init__
    print(upstream.model.__class__.__name__)
  File "/project/ocean/junweih/sandbox/espnet/tools/anaconda/envs/espnet/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1269, in __getattr__
    raise AttributeError("'{}' object has no attribute '{}'".format(
AttributeError: 'S3PRLUpstream' object has no attribute 'model'
# Accounting: time=11 threads=1
# Ended (code 1) at Mon Mar 13 11:57:11 EDT 2023, elapsed time 11 seconds
```

**Comment by sw005320** [comment]
@jwrh, thanks for the report!
@simpleoier, could you check this?

@jwrh, can you make a PR for this fix?

**Comment by jwrh** [comment]
Sure, creating PR now

**Comment by simpleoier** [comment]
@jwrh Thanks! You are right. I didn't notice that part was changed. I'll look at the PR.

**Comment by jwrh** [comment]
I think this has been solved by PR #5003, closing issue.
